libraries/tools/kotlin-gradle-plugin/build.gradle.kts

```diff
@@ -26,6 +27,12 @@ kotlin.sourceSets.all {
     languageSettings.optIn("org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi")
 }
 
+apiValidation {
+    publicMarkers.add("org.jetbrains.kotlin.gradle.ExternalKotlinTargetApi")
```

Comment:
are all APIs annotated with @ExternalKotlinTargetApi end up in kotlin-gradle-plugin-api artifact?

Reply:
No, those apis will be part of the `kotlin-gradle-plugin` artifact as those APIs are required to call into internal APIs of the implementation. So: `kotlin-gradle-plugin-api` already has binary compatibility validation, `kotlin-gradle-plugin` has not, but APIs marked with this annotation will have to be kept stable!